### PR TITLE
chore: Parallelize tests

### PR DESCRIPTION
**Pull request by rahulsom**
This will use all available cores to run tests, and parallelize by all - suite/class/method.

On machines with multiple cores, this will vastly improve test performance.

These times are on my M1 MBP with 10 (8P + 2E) cores. They were reported by maven on running `mvn verify`

Before parallelizing

```
Total time:  07:19 min
```

After parallelizing

```
Total time:  02:15 min
```

### Testing done

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]
Thanks!

**Comment by jtnord** [comment]
if it was this easy it would have been done in the `plugin-pom` and not in multiple plugins.

This is environmental setup that is best left for the person/thing that is running the tests.   Users can set `-DforkCount=12` should they wish before this change, after this change `-DforkCount=1` does not work.  Users with high CPU count machines now require high memory as they can no longer tune this,  e.g. a 3 year old i7 laptop chip with 16 logical processors requires 12GB of ram just to run the surefire forks defined here.

TL;DR users should set this in the CLI invocation, the jenkinsci `buildPlugin` could allow a way for this to be set for plugins running in CI.
